### PR TITLE
docs: split agent-api configuration options to a new doc

### PR DESCRIPTION
**Pull request by Qard**
Closes elastic/apm-agent-nodejs#473

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -0,0 +1,623 @@
+[[configuration]]
+
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html[elastic.co]
+endif::[]
+
+== Configuration
```
Comment (bmorelli25) [inline docs/configuration.asciidoc]:
```suggestion
=== Configuration
```

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+  // Set custom APM Server URL (default: http://localhost:8200)
+  serverUrl: '',
+
+  // Only activate the agent if it's running in production
+  active: process.env.NODE_ENV === 'production'
+})
+----
+
+[[service-name]]
+=== `serviceName`
```
Comment (bmorelli25) [inline docs/configuration.asciidoc]:
```suggestion
==== `serviceName`
```

File: docs/configuration.asciidoc (hunk 1 above)
Comment (Qard) [reply docs/configuration.asciidoc]: fixed.

File: docs/configuration.asciidoc (hunk 2 above)
Comment (Qard) [reply docs/configuration.asciidoc]: fixed.

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+
+The available configuration options are listed below.
+Most configuration options can be set either in the optional `options` object,
+by using environment variables,
+or via the <<agent-configuration-file,agent configuration file>>.
+Their equivalent environment variable name is listed together with each option.
+
+Environment variables will always take precedence over properties of the `options` object.
+
+The only required parameter is <<service-name,`serviceName`>>.
```
Comment (watson) [inline docs/configuration.asciidoc]: While not strictly related to this PR, I just stumbled upon this sentence that we probably overlooked when we implemented to fallback to `package.json`. So while we remember, we might as well update this text to say:

```suggestion
The only required parameter is <<service-name,`serviceName`>>.
However,
the agent will use the `name` from `package.json` by default if available.
```

Comment (Qard) [reply docs/configuration.asciidoc]: fixed.

**Comment by Qard** [comment]
Rebased.

File: docs/configuration.asciidoc (hunk 3 above)
Comment (watson) [reply docs/configuration.asciidoc]: 🤔 I don't see it anywhere. Maybe something went wrong during the rebase? 😬

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+
+Example usage configuring the agent to only be active in production:
+
+[source,js]
+----
+// Add this to the VERY top of the first file loaded in your app
+require('elastic-apm-node').start({
+  // Overwrite service name from package.json
```
Comment (watson) [inline docs/configuration.asciidoc]: I think the change got lost in the rebase:

```suggestion
  // Override service name from package.json
```

**Comment by Qard** [comment]
It's under the API reference section, so it's already covered by the link to that. We aren't currently linking to the sub-pages, should we?

**Comment by watson** [comment]
Ah yes, good point 🤔 By default we shouldn't link to sub-pages, but I'm a little on the fence now because this is such an important page...

**Comment by watson** [comment]
Let's just agree not to do it as part of this PR. If we later feel we need to link to it, it can go in a PR about reformatting the ToC 😃

**Comment by watson** [comment]
Not sure why codecov is failing. I think if the commit msg CI is fixed then codecov will probably fix it self

**Pull request merged**